src/GameSession.js

```diff
@@ -1,44 +1,101 @@
-const Squisher = require("./Squisher");
-const { generateName } = require('./common/util/name-generator');
+const Squisher = require('./Squisher');
+const { generateName } = require('./common/util');
 
 class GameSession {
-    constructor(game, res) {
+    constructor(game) {
         this.game = game;
-        this.squisher = new Squisher(res.width, res.height, game);
+        this.squisher = new Squisher(this.game);
         this.squisher.addListener(this);
-        this.players = {};
-        this.game.players = this.players;
-    }
-
-    async addPlayer(player) {
-        player.name = await generateName();
-         
-        player.addInputListener(this.squisher);
-        player.addStateListener(this);
-        this.players[player.id] = player;
-        const gameAssets = await this.squisher.getAssets();
-        if (gameAssets && gameAssets.length > 0) {
-            player.receiveUpdate(gameAssets);
+        this.renderWidth = this.game.constructor.metadata ? this.game.constructor.metadata().res.width : 1280;
+        this.renderHeight = this.game.constructor.metadata ? this.game.constructor.metadata().res.height : 720;
+    }
+
+    handleSquisherUpdate(squished) {
+        for (const playerId in this.game.players) {
+            this.game.players[playerId].receiveUpdate(squished);
+        }
+    }
+
+    addPlayer(player) {
+        generateName().then(playerName => {
+            player.name = playerName;
+            this.squisher.assetBundle && player.receiveUpdate(this.squisher.assetBundle);
+
+            player.receiveUpdate(this.squisher.squished);
+            this.game.addPlayer(player);
```

Comment:
this means the game can't contain a function called `addPlayer` we probably need to specify that somewhere or change the way we call "internal" functions on the game